keras_nlp/utils/text_generation.py

```diff
@@ -48,7 +37,42 @@ def validate_token_probability_fn(token_probability_fn, prompt):
         )
 
 
-def mask_tokens_after_end_token(prompt, max_length, end_token_id, pad_token_id):
+def _pad_prompts(prompt, max_length, pad_token_id):
+    # Maintain a mask tensor of shape `(batch_size,max_length)` which will
```

Comment:
Seems like this could just be the docstring? I would also phrase this more like...

"""
Pad prompt to mask_length and compute a mask for updates.

This utility will pad the (possibly ragged) prompt to max_length, and compute a mask where
input was originally set in the prompt, to avoid overwriting the original inputs when generating
new outputs.
"""